### PR TITLE
Add migration to fix parameter value types for alert conditions

### PR DESCRIPTION
**Pull request by bernd**
At one point at the beginning of the 2.2 development the values for integer parameters have been stored as strings in MongoDB. This has been fixed but existing broken documents need to be fixed.

### CODE REVIEW

File: graylog2-server/src/main/java/org/graylog2/migrations/V20170110150100_FixAlertConditionsMigration.java [hunk 1]
```diff
@@ -0,0 +1,150 @@
...
+
+                    // No need to convert anything if the field does not exist or is already an integer
+                    if (fieldValue == null || fieldValue instanceof Integer) {
+                        continue;
+                    }
+
+                    if (!(fieldValue instanceof String)) {
+                        LOG.warn("Field <{}> in alert condition <{}> of stream <{}> is not a string but a <{}>, not trying to convert it!",
+                                field, alertConditionId, streamId, fieldValue.getClass().getCanonicalName());
+                        return;
```
Comment (edmundoa) [inline graylog2-server/src/main/java/org/graylog2/migrations/V20170110150100_FixAlertConditionsMigration.java]: I think in this case we should only log the situation and continue converting other fields. This situation may happen with alert conditions created before 2.0. See https://github.com/Graylog2/graylog2-server/commit/ea4aa95fb609b56dcde4adc56cf311edf4ff3c0e for more information.

File: graylog2-server/src/main/java/org/graylog2/migrations/V20170110150100_FixAlertConditionsMigration.java [hunk 2]
```diff
@@ -0,0 +1,150 @@
...
+                for (String field : intFields) {
+                    final Object fieldValue = parameters.get(field);
+
+                    // No need to convert anything if the field does not exist or is already an integer
+                    if (fieldValue == null || fieldValue instanceof Integer) {
+                        continue;
+                    }
+
+                    if (!(fieldValue instanceof String)) {
+                        LOG.warn("Field <{}> in alert condition <{}> of stream <{}> is not a string but a <{}>, not trying to convert it!",
```
Comment (edmundoa) [inline graylog2-server/src/main/java/org/graylog2/migrations/V20170110150100_FixAlertConditionsMigration.java]: Could we also add the alert condition title if available?

File: graylog2-server/src/main/java/org/graylog2/migrations/V20170110150100_FixAlertConditionsMigration.java [hunk 3]
```diff
@@ -0,0 +1,150 @@
...
+                                field, alertConditionId, streamId, fieldValue.getClass().getCanonicalName());
+                        return;
+                    }
+
+                    final String stringValue = parameters.get(field, String.class);
+                    final Integer intValue = Ints.tryParse(stringValue);
+
+                    LOG.info("Converting value for field <{}> from string to integer in alert condition <{}> of stream <{}>", field, alertConditionId, streamId);
```
Comment (edmundoa) [inline graylog2-server/src/main/java/org/graylog2/migrations/V20170110150100_FixAlertConditionsMigration.java]: I would also add the alert condition title here.

File: graylog2-server/src/main/java/org/graylog2/migrations/V20170110150100_FixAlertConditionsMigration.java (hunk 1 above)
Comment (bernd) [inline graylog2-server/src/main/java/org/graylog2/migrations/V20170110150100_FixAlertConditionsMigration.java]: Oh, good catch! That should have been a `continue`.

**Comment by bernd** [comment]
Addressed all comments. Thanks!

**Pull request merged**